Sources/Amplitude/AMPStorage.m

```diff
@@ -92,51 +92,39 @@ + (void)storeEventAtUrl:(NSURL *)url event:(NSString *)event {
 }
 
 + (void)start:(NSString *)path {
-    NSString *contents = @"{ \"batch\": [";
+    NSString *contents = @"[";
     [[NSFileManager defaultManager] createDirectoryAtPath:[path stringByDeletingLastPathComponent] withIntermediateDirectories:YES attributes:NULL error:NULL];
     [[NSFileManager defaultManager] createFileAtPath:path contents:NULL attributes:NULL];
     [contents writeToFile:path atomically:YES encoding:NSUTF8StringEncoding error:NULL];
 }
 
-+ (void)finish:(NSString *)path {
-    NSFileManager *fm = [NSFileManager defaultManager];
-
-    if ([fm fileExistsAtPath:path]) {
-        NSString *fileEnding = @"]}";
-        NSData *endData = [fileEnding dataUsingEncoding:NSUTF8StringEncoding];
-
-        NSFileHandle *handle = [NSFileHandle fileHandleForWritingAtPath:path];
-        [handle seekToEndOfFile];
-        [handle writeData:endData];
-        [handle closeFile];
-    }
-}
-
 + (void)remove:(NSString *)path {
     NSFileManager *fm = [NSFileManager defaultManager];
     [fm removeItemAtPath:path error:NULL];
 }
 
 + (NSMutableArray *)getEventsFromDisk:(NSString *)path {
-    NSDictionary *json = [AMPStorage JSONFromFile:path];
-    NSArray *eventsArr = [json objectForKey:@"batch"];
+    NSArray *eventsArr = [AMPStorage JSONFromFile:path];
     return [eventsArr mutableCopy];
 }
 
-+ (NSDictionary *)JSONFromFile:(NSString *)path {
-    NSData *data = [NSData dataWithContentsOfFile:path];
-    if (data == nil) {
-        NSString *emptyData = @"{ \"batch\": []}";
-        data = [emptyData dataUsingEncoding:NSUTF8StringEncoding];
++ (NSArray*)JSONFromFile:(NSString *)path {
+    NSMutableString *content = [NSMutableString stringWithContentsOfFile:path encoding:NSUTF8StringEncoding error:nil];
+    if (content == nil) {
+        content = [[NSMutableString alloc] init];
+        [content appendString:@"["];
     }
 
-    NSError *err = nil;
-    NSDictionary *jsonAsDict = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:&err];
-    if (err != nil || jsonAsDict == nil) {
-        [AMPStorage finish:path];
-        data = [NSData dataWithContentsOfFile:path];
-        jsonAsDict = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:&err];
+    [content appendString:@"]"];
+    NSData *completeData = [content dataUsingEncoding:NSUTF8StringEncoding];
+    NSDictionary *jsonAsDict = [NSJSONSerialization JSONObjectWithData:completeData options:kNilOptions error:nil];
```

Comment:
The type should be updated as well?
I see `- WARN  | xcodebuild:  /Users/runner/work/Amplitude-iOS/Amplitude-iOS/Sources/Amplitude/AMPStorage.m:128:12: warning: incompatible pointer types returning 'NSDictionary *' from a function with result type 'NSArray *' [-Wincompatible-pointer-types]`

Reply:
Yea, I missed this one. Will change type to NSArray